metadata-io/src/main/java/com/linkedin/metadata/timeline/eventgenerator/DataProcessInstanceRunEventChangeEventGenerator.java

```diff
@@ -0,0 +1,99 @@
+import com.linkedin.dataprocess.DataProcessInstanceRunEvent;
+import com.linkedin.dataprocess.DataProcessRunStatus;
+import com.linkedin.metadata.entity.EntityService;
+import com.linkedin.metadata.timeline.data.ChangeCategory;
+import com.linkedin.metadata.timeline.data.ChangeEvent;
+import com.linkedin.metadata.timeline.data.ChangeOperation;
+import java.net.URISyntaxException;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+import static com.linkedin.metadata.Constants.*;
+
+
+public class DataProcessInstanceRunEventChangeEventGenerator
+    extends EntityChangeEventGenerator<DataProcessInstanceRunEvent> {
+  public DataProcessInstanceRunEventChangeEventGenerator(@Nonnull final EntityService entityService) {
+    super(entityService);
+  }
+
+  @Override
+  public List<ChangeEvent> getChangeEvents(@Nonnull Urn urn, @Nonnull String entity, @Nonnull String aspect,
+      @Nonnull Aspect<DataProcessInstanceRunEvent> from, @Nonnull Aspect<DataProcessInstanceRunEvent> to,
+      @Nonnull AuditStamp auditStamp) {
+    return computeDiffs(from.getValue(), to.getValue(), urn.toString(), auditStamp);
+  }
+
+  private List<ChangeEvent> computeDiffs(DataProcessInstanceRunEvent baseAspect,
+      DataProcessInstanceRunEvent targetAspect, String entityUrn, AuditStamp auditStamp) {
+    final DataProcessRunStatus baseStatus = getStatus(baseAspect);
+    final DataProcessRunStatus targetStatus = getStatus(targetAspect);
+
+    if (targetStatus != null && !targetStatus.equals(baseStatus)) {
+      String operationType = targetStatus.toString().equals("COMPLETE") ? "COMPLETED" : "STARTED";
+
+      return Collections.singletonList(ChangeEvent.builder()
+          .category(ChangeCategory.RUN)
+          .operation(ChangeOperation.valueOf(operationType))
+          .auditStamp(auditStamp)
+          .entityUrn(entityUrn)
+          .parameters(buildParamsMap(targetAspect, entityUrn))
+          .build());
+    }
+
+    return Collections.emptyList();
+  }
+
+  @Nullable
+  private DataProcessRunStatus getStatus(DataProcessInstanceRunEvent dataProcessInstanceRunEvent) {
+    return dataProcessInstanceRunEvent != null ? dataProcessInstanceRunEvent.getStatus() : null;
+  }
+
+  @Nonnull
+  private Map<String, Object> buildParamsMap(DataProcessInstanceRunEvent targetAspect, String entityUrnString) {
+    final Map<String, Object> paramsMap = new HashMap<>();
+    if (targetAspect.hasAttempt()) {
+      paramsMap.put(ATTEMPT_KEY, targetAspect.getAttempt());
+    }
+    if (targetAspect.hasResult()) {
+      paramsMap.put(RUN_RESULT_KEY, targetAspect.getResult().getType().toString());
+    }
+
+    Urn entityUrn;
+    try {
+      entityUrn = Urn.createFromString(entityUrnString);
+    } catch (URISyntaxException e) {
+      return paramsMap;
+    }
+
+    final DataProcessInstanceRelationships dataProcessInstanceRelationships =
```

Comment:
minor: I'd split this fetching logic into a separate method. Eventually, we might even want to extract into a service call

Reply:
Done!